Lib/tarfile.py

```diff
@@ -1727,6 +1727,8 @@ def __init__(self, name=None, mode="r", fileobj=None, format=None,
                                 # current position in the archive file
         self.inodes = {}        # dictionary caching the inodes of
                                 # archive members already added
+        self.uname_cache = {}   # Cached mappings of uid -> uname, gid -> gname
```

Comment:
I would split the comment (because gid -> gname is not the for this line, though one can understand). Maybe rename to `unames` and `gnames` since we only have `inodes` (it's not `inodes_cache`) or make them private (any reason to expose the attributes?)